### ISSUE TITLE
What you can do to contribute

### ISSUE THREAD

**Issue by Randium** [werewolves-devs/Werewolf_Bot#65]
In interpretation\ww_head.py, the message sent by the user is evaluated. With functions from interpretation\check.py, you can extract users, integers, roles and other things.
In roles_n_rules\functioncs.py, several functions are defined that take care of role actions. For example, the tanner will disguise their targets there, the fortune teller can see a target, etc.

The "weakness" of the functions in roles_n_rules\functions.py, is that it assumes the user is a nice guy, who has given the exact data it needs. To make sure this is the case, I'd like you to fix the commands in interpretation\ww_head.py. Extract the data that is necessary and pass it on to the function, and, if some needed data isn't given, send an error back to the user instead.
For example,  take a look at the assassin (line 337):
```
'''assassinate'''
# Assassin's command; kill a victim
if is_command(message,['assassinate','kill']) and user_role == "Assassin":
    target = check.users(message,1,True,True)
    if target == False:
        return [Mailbox().respond("**INVALID SYNTAX:**\nPlease make sure to mention a user.\n\n**Tip:** You can also mention their emoji!",True)]
    return [func.nightly_kill(user_id,target[0])]
```

First, a target to kill is extracted from the message. If it doesn't exist, an error is sent back. If it does, the target is passed on to the function, and the function's result is passed on to the user.

Notice how you don't need to care about game rules; you don't need to care about whether the assassin may kill themselves or not. You don't need to care if it is day or night and if the assassin has already killed someone. Stuff like that is taken care of in the function. You do, however, need to check if a user is a participant if that is what is needed for the function.
You only need to provide the data, or send an error back if there isn't enough (nor correct) data.

TLDR: Make the role functions in interpreation\ww_head.py.

Copied from the Discord server.

**Comment by romangraef** [comment]
another thing you could contribute, is reviewing pull requests if you are the repo admin

**Comment by Randium** [comment]
Closing this issue for now. We need to redefine how to contribute. Also, it's not like there's anyone even bothering these issues.
